src/DefinitionStorage.php

```diff
@@ -172,27 +174,18 @@ private function isResolvable(string $id, array $building): bool
                         if ($typeName === 'self') {
                             continue;
                         }
-                        $unionTypes[] = $typeName;
-                        if ($this->isResolvable($typeName, $building)) {
+
+                        if ($this->isResolvable($typeName, $building) ||
+                            ($this->delegateContainer !== null && $this->delegateContainer->has($typeName))) {
                             $isUnionTypeResolvable = true;
                             /** @infection-ignore-all Mutation don't change behaviour, but degrade performance. */
                             break;
                         }
                     }
 
-                    if (!$isUnionTypeResolvable) {
```

Comment:
????

Reply:
Yeah. "Optimization". Tests pass so "why not" :)